board.py

```diff
@@ -0,0 +1,268 @@
+import uuid
+import pickle
+import os
+import sys
+from user import *
+from chirp import *
+
+class Birdyboard:
+    def __init__(self, boardfile, userfile):
+        """
+        On class creation this function opens corrisponding files and loads the data to chosen variables
+        """
+        self.currentUser = None
+        self.privateRecip = None
+        self.userPrivateChirps = {}
+        self.boardfile = boardfile
+        self.userfile = userfile
+        self.listofChirps = []
+        self.clearScreen()
+        try:
+            self.deserializeChirps(boardfile)
+        except EOFError:
+            self.chirps = {}
+            self.chirps['private'] = {}
+            self.chirps['public'] = {}
+        try:
+            self.deserializeUsers(userfile)
+        except EOFError:
+            self.users = []
+
+    def deserializeChirps(self, filename):
+        """Deserialize chirp file"""
+        try:
+            with open(filename, 'rb+') as f:
+                self.chirps = pickle.load(f)
+        except FileNotFoundError:
+            self.chirps = {}
+            self.chirps['private'] = {}
+            self.chirps['public'] = {}
+        return self.chirps
+
+    def deserializeUsers(self, filename):
+        """Deserialize user file"""
+        try:
+            with open(filename, 'rb+') as f:
+                self.users = pickle.load(f)
+        except FileNotFoundError:
+            self.users = []
+        return self.users
+
+    def serializeUsers(self, filename):
+        """Serialize user file"""
+        with open(filename, 'wb+') as f:
+            pickle.dump(self.users, f)
+
+    def serializeMessages(self, filename):
+        """Serialize chirp file"""
+        with open(filename, 'wb+') as f:
+            pickle.dump(self.chirps, f)
+
+    def menu(self):
+        """Function to show the main menu with 6 options"""
+
+        print("""
+<<<<<<<<<<<<<<<<<<<<>>>>>>>>>>>>>>>>>>>>
+<<<         <<<Birdyboard>>>         >>>
+<<<<<<<<<<<<<<<<<<<<>>>>>>>>>>>>>>>>>>>>
+1. New User Account
+2. Select User
+3. View Chirps
+4. New Public Chirp
+5. New Private Chirp
+6. Exit
+        """)
+        choice = input('> ')
+        if choice == '1':
+            self.clearScreen()
+            name = input("Enter Full Name: \n> ")
+            username = input("Enter User Name: \n> ")
+            self.newUser(name, username)
+            self.clearScreen()
+            self.menu()
+        elif choice == '2':
+            self.clearScreen()
+            self.printAllUsers()
+            userChoice = input('Select User \n> ')
+            self.selectUser(int(userChoice))
```

Comment:
On line 119 there's an else statement if the user types in anything other
than 1-6


On Mon, Aug 8, 2016 at 9:48 AM, Ryan Tanay <notifications@github.com> wrote:

> In board.py
> <https://github.com/nss-day-cohort-13/birdyboard-hannahhall/pull/1#discussion_r73888694>
> :
>
> > +5. New Private Chirp
> > +6. Exit
> > +        """)
> > +        choice = input('> ')
> > +        if choice == '1':
> > +            self.clearScreen()
> > +            name = input("Enter Full Name: \n> ")
> > +            username = input("Enter User Name: \n> ")
> > +            self.newUser(name, username)
> > +            self.clearScreen()
> > +            self.menu()
> > +        elif choice == '2':
> > +            self.clearScreen()
> > +            self.printAllUsers()
> > +            userChoice = input('Select User \n> ')
> > +            self.selectUser(int(userChoice))
>
> What would happen if I misunderstand the menu and type in the user name?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nss-day-cohort-13/birdyboard-hannahhall/pull/1/files/301ee5395630f6c6575762d9fc7aa239d74b178c#r73888694>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARbGAIoTlfNNc3I_ZPWuyv8iw63-3ftNks5qd0HNgaJpZM4JZ8tV>
> .
>